### PR TITLE
Introducing and handling a termination message

### LINKED ISSUE #5: Send terminate message to external simulation at simulation end
Currently, the external simulation does not receive a terminate message at the end of a simulation but is just simply not triggered anymore at some point.

Niklas, Chris, and I had a short discussion about this and thought it might make sense to adapt this accordingly.

tbd

### PR DESCRIPTION
**Pull request by sebastian-peter**
Resolves #5

Plus enabling tests w/ gradle and Jenkins
Plus implementation of tests for ExtSimulation

### CODE REVIEW

File: src/test/groovy/edu/ie3/simona/api/simulation/ExtSimulationSpec.groovy [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+            extSimData.queueExtMsg(new SimTerminated())
+            def finishedActual = handleMessage.invoke(extSim)
+
+        then:
+            finishedActual == true
+            testProbe.expectNoMessage()
+            verify(extSim).terminated() // terminated() has been called exactly once
+    }
+
+    class UnknownMessage implements ExtSimMessage {}
```
[the remark this answers is not in the document]
Comment (sebastian-peter) [reply src/test/groovy/edu/ie3/simona/api/simulation/ExtSimulationSpec.groovy:84]: @sonatype-lift ignore

**Review by ckittl — CHANGES_REQUESTED** [review]
Generally looks good to me!

However, only point of discussion is, if we wouldn't like to see the termination as a pro-active step rather then "uh, someone else has terminated, better do that, too". In that regard: Do we need a feedback about successful termination?

**Comment by ckittl** [comment]
... an please add your contributions to CHANGELOG.md

**Comment by sebastian-peter** [comment]
@ckittl introduced indication whether main simulation was successful, and a `TerminationCompleted` message that is sent back to main simulation when the termination process in ext sim has finished (as discussed)

**Review by ckittl — APPROVED** [review]

**Pull request merged**